lib/bango/views/billing.py

```diff
@@ -0,0 +1,144 @@
+    configs = {
+        'APPLICATION_CATEGORY_ID': '18',
+        'APPLICATION_SIZE_KB': data.pop('application_size'),
+        # Tell Bango to use our same transaction expiry logic.
+        # However, we pad it by 60 seconds to show a prettier Mozilla user
+        # error in the case of a real timeout.
+        'BILLING_CONFIGURATION_TIME_OUT': settings.TRANSACTION_EXPIRY + 60,
+        'REDIRECT_URL_ONSUCCESS': data.pop('redirect_url_onsuccess'),
+        'REDIRECT_URL_ONERROR': data.pop('redirect_url_onerror'),
+        'REQUEST_SIGNATURE': sign(data['externalTransactionId']),
+    }
+    user_uuid = data.pop('user_uuid')
+    if settings.SEND_USER_ID_TO_BANGO:
+        configs['MOZ_USER_ID'] = user_uuid
+        log.info('Sending MOZ_USER_ID {uuid} for transaction {tr}'
+                 .format(uuid=user_uuid, tr=data['externalTransactionId']))
+    if settings.BANGO_ICON_URLS:
+        icon_url = data.pop('icon_url', None)
+        if icon_url:
+            configs['APPLICATION_LOGO_URL'] = icon_url
+
+    for k, v in configs.items():
+        opt = billing.factory.create('BillingConfigurationOption')
+        opt.configurationOptionName = k
+        opt.configurationOptionValue = v
+        config.BillingConfigurationOption.append(opt)
+
+    data['configurationOptions'] = config
+    return data
+
+
+@api_view(['POST'])
+def billing(request):
+    """
+    Call the Bango API to begin a payment transaction.
+
+    The resulting billingConfigId can be used on the query
+    string in a URL to initiate a user payment flow.
+
+    We are able to configure a few parameters that come
+    back to us on the Bango success URL query string.
+    Here are some highlights:
+
+    **config[REQUEST_SIGNATURE]**
+        This arrives as **MozSignature** in the redirect query string.
+
+    **externalTransactionId**
+        This is set to solitude's own transaction_uuid. It arrives
+        in the redirect query string as **MerchantTransactionId**.
+    """
+
+    view = BangoResource()
+    try:
+        serial, form = view.process(
+            serial_class=SellerProductBangoOnly,
+            form_class=CreateBillingConfigurationForm,
+            request=request)
+    except ProcessError, exc:
+        return exc.response
+
+    transaction_uuid = form.bango_data['externalTransactionId']
+
+    try:
+        data = prepare(form)
+        resp = view.client('CreateBillingConfiguration', data)
+    except BangoError:
+        log.error('Error on createBillingConfiguration, uuid: {0}'
+                  .format(transaction_uuid))
+        raise
+
+    response_data = {
+        'responseCode': resp.responseCode,
+        'responseMessage': resp.responseMessage,
+        'billingConfigurationId': resp.billingConfigurationId
+    }
+
+    log.info('Sending trans uuid %s from Bango config %s'
+             % (transaction_uuid,
+                response_data['billingConfigurationId']))
```

Comment:
Well, the DRF change is such a beast I guess it's a nit not worth worrying about ;)